src/main/java/interview/project/Github/GithubProxyBranchService.java

```diff
@@ -0,0 +1,41 @@
+package interview.project.Github;
+
+import interview.project.Github.data.GithubBranch;
+import interview.project.Github.data.ProxyBranch;
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@Service
+@AllArgsConstructor
+public class GithubProxyBranchService {
+    private RestTemplate restTemplate;
+
+    private HttpEntity<String> authorizationTokenHttpEntity;
+
+    public List<ProxyBranch> listBranchesForRepository(String username, String repositoryName) {
+        final String url = String.format("https://api.github.com/repos/%s/%s/branches", username, repositoryName);
+
+        GithubBranch[] branches;
```

Comment:
This is redundant, simply return from the "try" block instead of defining mutable, nullable variable